lib/toopher_api.rb

```diff
@@ -244,26 +435,242 @@ class AuthenticationStatus
+class UserTerminals
+  def initialize(raw)
+    @raw = raw
+  end
+
+  # Check on the status of a user terminal
+  #
+  # @param [String] terminal_id A unique string identifier generated and returned by the Toopher web service that is used to identify this user terminal.
+  def get_by_id(terminal_id)
+    UserTerminal.new(@raw.get('user_terminals/' + terminal_id))
+  end
+
+  # Create a new user terminal
+  #
+  # @param [String] username The human recognizable user name of the user you wish to associate this user terminal.
+  # @param [String] terminal_name The human recognizable terminal name.
+  def create(username, terminal_name, requester_terminal_id, **kwargs)
+    params = {
+      :user_name => username,
+      :name => terminal_name,
+      :name_extra => requester_terminal_id
+    }
+    params.merge!(kwargs)
+    UserTerminal.new(@raw.post('user_terminals/create', params))
+  end
+
+end
+
+# Contains information about a particular user terminal
+class UserTerminal
+  # @!attribute id
+  #   @return [String] A unique string identifier generated and returned by the Toopher web service that is used to identify this user terminal. It can be used to request status information for the user terminal.
+  attr_accessor :id
+
+  # @!attribute name
+  #   @return [String] The human recognizable terminal name associated with the given id.
+  attr_accessor :name
+
+  # @!attribute name_extra
+  #   @return [String] A string to help differentiate identically named terminals.
+  attr_accessor :name_extra
+
+  # @!attribute user
+  #   @return [User] Contains information about the User associated with this UserTerminal
+  attr_accessor :user
+
+  # @!attribute raw
+  #   @return [hash] The raw data returned from the Toopher API
+  attr_accessor :raw
+
+  def initialize(json_obj)
+    @user = User.new(json_obj['user'])
+    update(json_obj)
+  end
+
+  def refresh_from_server(api)
+    result = api.advanced.raw.get('user_terminals/' + @id)
+    update(result)
+  end
+
+  private
+
+  def update(json_obj)
+    @id = json_obj['id']
+    @name = json_obj['name']
+    @name_extra = json_obj['name_extra']
+    @user.send(:update, json_obj['user'])
+    @raw = json_obj
+  end
+end
+
+# Contains advanced ToopherApi methods associated with users
+class Users
+  def initialize(raw)
+    @raw = raw
+  end
+
+  # Check on the status of a user
+  #
+  # @param [String] user_id A unique string identifier generated and returned by the Toopher web service that is used to identify this user.
+  def get_by_id(user_id)
+    User.new(@raw.get('users/' + user_id))
+  end
+
+  # Check on the status of a user
+  #
+  # @param [String] name The human recognizable user name.
+  def get_by_name(username)
+    params = { :name => username }
+    users = @raw.get('users', params)
+    if users.count > 1
+      raise ToopherApiError, 'Multiple users with name = #{username}'
+    elsif users.count == 0
+      raise ToopherApiError, 'No users with name = #{username}'
```

Comment:
Double-quotes should be used for string interpolation.
`"Multiple users with name = #{username}"`
`"No users with name = #{username}"`